autogluon/utils/tabular/ml/models/lr/lr_model.py

```diff
@@ -0,0 +1,206 @@
+
+            if len(feature_types['continuous']) > 0:
+                pipeline = Pipeline(steps=[
+                    ('generator', NumericDataPreprocessor(cont_cols=feature_types['continuous'])),
+                    ('imputer', SimpleImputer(strategy=self.params['proc.impute_strategy'])),
+                    ('scaler', StandardScaler())
+                ])
+                transformer_list.append(('cont', pipeline))
+
+            if len(feature_types['skewed']) > 0:
+                pipeline = Pipeline(steps=[
+                    ('generator', NumericDataPreprocessor(cont_cols=feature_types['skewed'])),
+                    ('imputer', SimpleImputer(strategy=self.params['proc.impute_strategy'])),
+                    ('quantile', QuantileTransformer(output_distribution='normal')),  # Or output_distribution = 'uniform'
+                ])
+                transformer_list.append(('skew', pipeline))
+
+            self.pipeline = FeatureUnion(transformer_list=transformer_list)
+            logger.log(15, " - fitting pre-processing pipeline")
+            self.pipeline.fit(X)
+
+        logger.log(15, " - transforming inputs using pre-processing pipeline")
+        X = self.pipeline.transform(X)
+        logger.log(15, " - output shape %s" % str(X.shape))
+
+        return X
+
+    def _set_default_params(self):
+        default_params = {
+            'model_type': 'LR',
+            'C': 1,
+            'vectorizer_dict_size': 75000,
+            'proc.ngram_range': (1, 5),
+            'proc.skew_threshold': 0.99,
+            'proc.impute_strategy': 'median',  # strategy argument of sklearn.SimpleImputer() used to impute missing numeric values
+        }
+        for param, val in default_params.items():
+            self._set_default_param_value(param, val)
+
+    def _get_default_searchspace(self, problem_type):
+        spaces = {
+            # 'C': Real(lower=1e-4, upper=1e5, default=1),
+            # 'tokenizer': Categorical('split', 'sentencepiece')
+        }
+        return spaces
+
+    def fit(self, X_train, Y_train, X_test=None, Y_test=None, time_limit=None, **kwargs):
+        hyperparams = self.params.copy()
+
+        self.cat_one_hot = get_one_hot_features(X_train)
+
+        # See solver options here: https://scikit-learn.org/stable/modules/generated/sklearn.linear_model.LogisticRegression.html
+        if self.problem_type == BINARY:
+            Y_train = Y_train.astype(int).values
+            solver = 'lbfgs'  # TODO use liblinear for smaller datasets
+        elif self.problem_type == MULTICLASS:
+            solver = 'saga'  # another option is lbfgs
+        else:
+            solver = 'lbfgs'
+
+        X_train = self.preprocess(X_train, is_train=True, vect_max_features=hyperparams['vectorizer_dict_size'])
+
+        # TODO: get seed from seeds provider
+        self.model = LogisticRegression(C=hyperparams['C'], random_state=17, solver=solver, n_jobs=-1)
```

Comment:
Only set self.model after .fit(), .fit() should return self

self.model = model.fit(X_train, Y_train)

Example is in KNN

Reply:
Done